Magento2/Sniffs/Commenting/ClassAndInterfacePHPDocFormattingSniff.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * Copyright © Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento2\Sniffs\Commenting;
+
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+
+/**
+ * Detects PHPDoc formatting for classes and interfaces.
+ */
+class ClassAndInterfacePHPDocFormattingSniff extends ConstantsPHPDocFormattingSniff implements Sniff
```

Comment:
`ClassAndInterfacePHPDocFormattingSniff` should not extend `ConstantsPHPDocFormattingSniff `. Let's move `providesMeaning` to the separate class (aka helper) and call this method from there (it can be even `static`). In future some common methods related to Doc Blocks formatting can be places there.